app/models/concerns/profile_searching.rb

```diff
@@ -35,8 +35,22 @@ module ProfileSearching
       where(external: external)
     }
     scope :has_test_results, lambda { |has_test_results = true|
-      test_results = TestResult.select(:profile_id).distinct
-      has_test_results && where(id: test_results) || where.not(id: test_results)
+      # the use of default scope is to bypass Pundit
+      # and avoid the situation with bind parameters
+      # that must have a different order in the resulting query
+      # of a scoped_search
+      policy_test_results = default_scoped.joins(policy_object: :test_results)
+
+      internal = arel_table[:external].eq(false).and(
+        arel_table[:id].in(policy_test_results.select(:id).arel)
+      )
```

Comment:
I'm not sure this is quite right (notice the change from id to profile_id below). Also, it could be written like this:

```suggestion
      internal = external(false).where(id: policy_test_results.select(:profile_id))
```

Reply:
You can translate it into:
```
Profile.joins(policy_object: :test_results).select(:id)
# or
Profile.joins(policy_object: :test_results).pluck(:id)
```
It leverages the INNER JOIN. It should yield profile ids have a policy and their policy profiles have a test result.

Here is the subquery:
```
SELECT "profiles"."id"
FROM "profiles"
INNER JOIN "policies" ON "policies"."id" = "profiles"."policy_id"
INNER JOIN "profiles" "profiles_policies_join" ON "profiles_policies_join"."policy_id" = "policies"."id"
INNER JOIN "test_results" ON "test_results"."profile_id" = "profiles_policies_join"."id"
```

Reply:
Ok, I see. Thanks! It seems correct.